### DISCUSSION TITLE
Does using the --inspect flag have any performance implications?

### DISCUSSION THREAD

**Discussion by kylecombes** [nodejs/help#4878] [Q&A]
I'm configuring VS Code debugging for a Next.js app, and my initial thought was to just change our `dev` script in our `package.json` to be `NODE_OPTIONS='--inspect' next dev`. This would obviously mean that anytime a dev ran our app, it would run with this flag, even if the dev didn't need to debug the Node process.

I'm trying to determine whether using this flag has any performance implications, but the only "answers" I've found to this question are the two "feelings" expressed in [this Stack Overflow question](https://stackoverflow.com/questions/45379080/does-the-node-inspect-flag-have-any-notable-performance-impact). Can anyone here provide a more evidence-based opinion?

**Comment by Trott** [comment]
@nodejs/diagnostics

**Comment by Trott** [comment]
If you don't have something you can benchmark yourself, and since your request is specifically for data over general sense, you might want to take a look at `doc/guides/writing-and-running-benchmarks.md` and note especially that `createBenchmark()` takes an `options` argument that allows you to specify command-line flags. You can use that to modify an existing benchmark to run with `--inspect` and then run it without `--inspect` and compare the results.

**Comment by devsnek** [comment]
enabling the inspector does not hurt perf but *connecting* to it tiers down all the wasm, and trying to step or pause js causes that js to tier down.
